bugsnag/src/main/java/com/bugsnag/callbacks/DeviceCallback.java

```diff
@@ -45,18 +45,15 @@ private static String lookupHostname() {
             return hostname;
         }
 
-        // Resort to dns hostname lookup.
-        // Look up the hostname in a daemon thread, with a timeout of HOSTNAME_LOOKUP_TIMEOUT.
-        // This is to prevent applications hanging waiting for the dns lookup to resolve
+        // Resort to dns hostname lookup with a timeout of HOSTNAME_LOOKUP_TIMEOUT,
+        // as this can potentially take a very long time to resolve
         FutureTask<String> future = new FutureTask<String>(new Callable<String>() {
             @Override
             public String call() throws UnknownHostException {
                 return InetAddress.getLocalHost().getHostName();
             }
         });
-        Thread resolverThread = new Thread(future, "Hostname Resolver");
-        resolverThread.setDaemon(true);
-        resolverThread.start();
+        future.run();
```

Comment:
This will need to be threaded. `future.run()` will wait for the hostname to be resolved before returning, which means that `future.get(HOSTNAME_LOOKUP_TIMEOUT...` will return immediately (and never timeout).